src/providers/test-result/test-result.ts

```diff
@@ -14,7 +15,18 @@ export class TestResultProvider {
     private faultCountProvider: FaultCountProvider,
   ) {}
 
-  calculateCatBTestResult = (testData: CatBUniqueTypes.TestData): Observable<ActivityCode> => {
+  calculateTestResult = (category: string,
+                         testData: CatBUniqueTypes.TestData| CatBEUniqueTypes.TestData): Observable<ActivityCode> => {
+    if (category === TestCategory.B) {
```

Comment:
can this be a switch on TestCategory.

alternatively could we not use the functions directly where they are used instead of having this extra layer of abstraction. I can't see what use we are getting out of it

Reply:
Changed to a switch.